### PR TITLE
Initial documentation

### PR DESCRIPTION
**Pull request by crezvoy**
- Add basic documentation to the readme
- fix examples
- remove refs to DM.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,95 @@
...
+An Airflow plugin and provider to launch and monitor Spark
+applications on the [Ocean for
+Spark](https://spot.io/products/ocean-apache-spark/).
 
-# Compatibility
+## Compatibility
 
-# Installation
+`ocean-spark-airflow-provider` is compatible with both airflow 1 and
+airflow 2. it is detected as an airflow plugin by airlfow 1 and up,
```
Comment (jrj-d) [inline README.md:10]:
```suggestion
airflow 2. it is detected as an airflow plugin by airflow 1 and up,
```
Why "and up"? Do you mean it is detected as a plugin and as a provider by Airflow 2?

Comment (jrj-d) [reply README.md:10]: Also, please add a capital "A" to all instances of Airflow in the text

File: README.md [hunk 2]
```diff
@@ -1,11 +1,95 @@
...
+
+## Usage
+
+For general usage of Ocean for Spark, refer to the [official
+documentation](https://docs.spot.io/ocean-spark/getting-started/?id=get-started-with-ocean-for-apache-spark).
+
+### Setting up the connection
+
+In the connection menu, register a new connection of type **Ocean For
+Spark**, the default connection name is `ocean_spark_default`.you will
```
Comment (jrj-d) [inline README.md:29]:
```suggestion
Spark**. The default connection name is `ocean_spark_default`. You will
```

File: README.md [hunk 3]
```diff
@@ -1,11 +1,95 @@
...
+        "executor": {
+            "cores": 2,
+            "memory": "1g",
+            "labels": {"version": "3.0.0"},
+            "instances": 2,
+        },
+    },
+)
+```
+
```
Comment (jrj-d) [inline README.md:90]: Can you add a link to your example folders?

File: README.md [hunk 4]
```diff
@@ -1,11 +1,95 @@
...
+   clusters](https://docs.spot.io/ocean-spark/product-tour/manage-clusters),
+   or by using the [Get Cluster
+   List](https://docs.spot.io/api/#operation/OceanSparkClusterList) in
+   the API.
+ - [A Spot
+   token](https://docs.spot.io/administration/api/create-api-token?id=create-an-api-token)
+   to interact with Spot API.
+ 
+![connection setup dialog](./images/connection_setup.png) 
+
```
Comment (jrj-d) [inline README.md:44]: Can you add a sentence like:
```
You will need to create a separate connection for every Ocean Spark cluster that you plan to use with Airflow.
In the `OceanSparkOperator`, you can select which Ocean Spark connection to use with the `connection_name` argument (defaults to `ocean_spark_default`).
```

File: ocean_spark/hooks.py [hunk 5]
```diff
@@ -50,7 +56,6 @@ def __init__(
         self.conn = self.get_connection(ocean_spark_conn_id)
         self.token = self.conn.password
         self.cluster_id = self.conn.host
-        self.account_id = self.conn.login
```
Comment (jrj-d) [inline ocean_spark/hooks.py:53]: How is the account Id determined? I see a potential issue

File: deploy/airflow1/example_dags/single_task_dag.py [hunk 6]
```diff
@@ -16,9 +16,12 @@
     dag=dag,
     config_overrides={
         "type": "Scala",
-        "sparkVersion": "3.0.0",
-        "mainApplicationFile": "local:///opt/spark/examples/jars/spark-examples_2.12-3.0.0.jar",
+        "sparkVersion": "3.2.0",
+        "interactive": False,
```
Comment (jrj-d) [inline deploy/airflow1/example_dags/single_task_dag.py:20]: Can you remove references to the `interactive` field everywhere?

File: README.md (hunk 1 above)
Comment (crezvoy) [reply README.md]: > Why "and up"? Do you mean it is detected as a plugin and as a provider by Airflow 2?

Yes, its is detected both as a plugin and a provider by Airflow 2. Airflow does not seem to mind, I think the only consequence is that one can import the operator from `airflow.operators.` (plugin) and from `ocean_spark.operators` (provider).

File: deploy/airflow1/example_dags/single_task_dag.py (hunk 6 above)
Comment (crezvoy) [reply deploy/airflow1/example_dags/single_task_dag.py]: done

File: README.md [hunk 7]
```diff
@@ -1,11 +1,102 @@
...
+    task_id="compute-pi",
+    dag=dag,
+    config_overrides={
+        "type": "Scala",
+        "sparkVersion": "3.2.0",
+        "image": "gcr.io/datamechanics/spark:platform-3.2-latest",
+        "imagePullPolicy": "IfNotPresent",
+        "mainClass": "org.apache.spark.examples.SparkPi",
+        "mainApplicationFile": "local:///opt/spark/examples/jars/examples.jar",
+        "arguments": ["100000"],
```
Comment (jystephan) [inline README.md:79]: too large

File: README.md [hunk 8]
```diff
@@ -1,11 +1,102 @@
...
+```python
+from airflow import __version__ as airflow_version
+if airflow_version.starts_with("1."):
+    # Airflow 1, import as plugin
+    from airflow.operators.ocean_spark import OceanSparkOperator
+else:
+    # Airflow 2
+    from ocean_spark.operators import OceanSparkOperator
+    
+# DAG creation ...
```
Comment (jystephan) [inline README.md:66]: nit: Remove "..."

File: README.md [hunk 9]
```diff
@@ -1,11 +1,102 @@
...
+            "cores": 1,
+            "coreLimit": "1200m",
+            "memory": "1g",
+            "labels": {"version": "3.0.0"},
+        },
+        "executor": {
+            "cores": 2,
+            "memory": "1g",
+            "labels": {"version": "3.0.0"},
+            "instances": 2,
```
Comment (jystephan) [inline README.md:90]: let's use 1 instance by default

File: README.md [hunk 10]
```diff
@@ -1,11 +1,102 @@
...
+        "type": "Scala",
+        "sparkVersion": "3.2.0",
+        "image": "gcr.io/datamechanics/spark:platform-3.2-latest",
+        "imagePullPolicy": "IfNotPresent",
+        "mainClass": "org.apache.spark.examples.SparkPi",
+        "mainApplicationFile": "local:///opt/spark/examples/jars/examples.jar",
+        "arguments": ["100000"],
+        "driver": {
+            "cores": 1,
+            "coreLimit": "1200m",
```
Comment (jystephan) [inline README.md:82]: We should remove the coreLimit and memory fields, once we have the smart default logic implemented by @geoffroynicolas. 

The good practice we want to encourage is for users to only pass the cores field, and also an instanceSelector. 

If for now the coreLimit and memory fields are required, let's keep them.

File: README.md [hunk 11]
```diff
@@ -1,11 +1,102 @@
...
+    },
+)
+```
+
+more examples are available for [Airflow 1](./deploy/airflow1/example_dags) and [Airflow 2](./deploy/airflow2/dags).
+
+## Test locally
+
+You can test the plugin locally using the docker compose setup in this
+repositor. Run `make serve_airflow2` at the root of the repository to
```
Comment (jystephan) [inline README.md:101]: repository

**Review by jystephan — COMMENTED** [review]
Thanks! Minor comments inlined.

File: README.md (hunk 3 above)
Comment (crezvoy) [reply README.md:95]: set all spark examples to 10000

File: README.md (hunk 10 above)
Comment (crezvoy) [reply README.md]: I removed those fields I don't think they're necessary. they're not in the example I use for testing

File: README.md [hunk 12]
```diff
@@ -1,11 +1,97 @@
...
+`ocean-spark-airflow-provider` is compatible with both Airflow 1 and
+Airflow 2. it is detected as an Airflow plugin by Airflow 1 and up,
+and as a provider by Airflow 2.
 
-# Usage
 
-# Example
+## Installation
+
+TODO(crezvoy): setup pypi repo
```
Comment (jrj-d) [inline README.md:16]: Remove todo

**Pull request merged**